phpBB/phpbb/search/backend/base.php

```diff
@@ -329,7 +336,8 @@ public function create_index(int &$post_counter = 0): ?array
 		$starttime = microtime(true);
 		$row_count = 0;
 
-		while (still_on_time() && $post_counter <= $max_post_id)
+		$still_on_time = PHP_SAPI === 'cli' ? true : still_on_time();
+		while ($still_on_time && $post_counter < $max_post_id)
```

Comment:
```suggestion
		while ($still_on_time && $post_counter <= $max_post_id)
```
as you can see, I haven't done that in PR : 
https://github.com/phpbb/phpbb/pull/6348/files#diff-32ffb04769629c7e08d598c9524dd983c482542549a5214157d286af09f518b6L386

Reply:
Yes, but if I dont change it I receive a timeout because of an infinite loop (that is why I ask why didn't happen to you or if i am doing anything wrong now)

Reply:
> if I dont change it I receive a timeout because of an infinite loop

That makes sense as with CLI you have `$still_on_time` always `true` and at the end of the process `$post_counter` is always equal to `$max_post_id`, so with `$post_counter <= $max_post_id` the loop is infinite. Probably the edge case of `$post_counter == $max_post_id` should be handled separately, otherwise the last post in the queue won't be indexed.

Reply:
If `$post_counter == $max_post_id`, if I am not wrong, all posts have been already indexed, because for each iteration, the post that are indexed are those that go after $post_counter without including it

Reply:
https://github.com/phpbb/phpbb/pull/5439/files/a7670ae74aee724722ec3e56579d2adf06e847c7#diff-32ffb04769629c7e08d598c9524dd983c482542549a5214157d286af09f518b6R360-R363

this lines should have taken care of an infinite loop, or am I missing something.

Reply:
Yes, but only when using CLI, because the bucle do only one iteration, but not with the web interface.
(I am thinking that change it with a do-while should be more clean)

Reply:
Got the issue,
the `still_on_time()` needs to be called every time the `while(...)` block is executed to check for `still_on_time()`,
but here it's done only once out side the `while(...)` block.
```suggestion
		while ((PHP_SAPI === 'cli' ? true : still_on_time()) && $post_counter <= $max_post_id)
```
.
I am still checking it that loop needs `<=` or not.
I'll report this toomoro.

Best regards 👍

Reply:
the `while(...)` loop does not require `<=`,
but require `still_on_time()` for each run of `while(...)` block.

Final changes:
```suggestion
		while ((PHP_SAPI === 'cli' ? true : still_on_time()) && $post_counter < $max_post_id)
```